packages/backend-modules/job-queue/lib/queue.ts

```diff
@@ -0,0 +1,95 @@
+import PgBoss from 'pg-boss'
+import { Worker, WorkerJobArgs, WorkerQueueName } from './types'
+
+export class Queue {
+  protected readonly pgBoss: PgBoss
+  protected workers = new Map<WorkerQueueName<Worker<any>>, Worker<any>>()
+
+  constructor(options: PgBoss.ConstructorOptions) {
+    this.pgBoss = new PgBoss(options)
+
+    this.pgBoss.on('error', (error) => {
+      console.error('[JobQueue]: %s', error)
+    })
+  }
+
+  registerWorker(worker: new (pgBoss: PgBoss) => Worker<any>): Queue {
+    const workerInstance = new worker(this.pgBoss)
+    this.workers.set(workerInstance.queue, workerInstance)
+    return this
+  }
+
+  async start() {
+    await this.pgBoss.start()
+  }
+
+  async stop() {
+    await this.pgBoss.stop({
+      timeout: 20000,
+    })
+
+    await new Promise((resolve) => {
+      this.pgBoss.once('stopped', () => resolve(null))
+    })
+  }
+
+  async getJobById(id: string) {
+    return this.pgBoss.getJobById(id)
+  }
+
+  async clearStorage() {
+    return this.pgBoss.clearStorage()
+  }
+
+  async startWorkers() {
+    const workers: Promise<string>[] = []
+
+    for (const worker of this.workers.values()) {
+      workers.push(this.pgBoss.work(worker.queue, worker.perform))
+    }
```

Comment:
why do we not directly return 

`this.pgBoss.work(worker.queue, worker.perform)`?

Reply:
if we would await each worker, the processing won't happen concurrently because each loop iteration would wait for the worker to start before starting the next one. By using `Promise.all` the workers get started concurrently and we can await all Promises at once.